### PR TITLE
Anonymous items

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: *(direct link to the issue)*

Clarify one of the larger confusions about how the Registry system is meant to be used,
about so-called "Transparent" items.

Some other changes of terminology too.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Review by crsib — APPROVED** [review]
Much better naming indeed!

**Comment by crsib** [comment]
I think it can go straight to master

File: libraries/lib-registries/Registry.cpp [hunk 1]
```diff
@@ -377,28 +391,30 @@ auto CollectedItems::MergeWithExistingItem(
...
             // Subordinate items from one of the groups will be merged in
             // another call to MergeItems at a lower level of path.
-            // Note, however, that at most one of the two should be other
-            // than a plain grouping item; if not, we must lose the extra
+            // Note, however, that at most one of the two should be a
+            // strongly ordered item; if not, we must lose the extra
             // information carried by one of them.
-            bool pCollectionGrouping = pCollectionGroup->Transparent();
-            auto pRegistryGrouping = pRegistryGroup->Transparent();
+            bool pCollectionGrouping =
```
Comment (saintmatthieu) [inline libraries/lib-registries/Registry.cpp:398]: Could be const

Comment (saintmatthieu) [inline libraries/lib-registries/Registry.cpp:398]: Why the prefix `p` ? I imagined this was reserved to pointers.

File: libraries/lib-registries/Registry.h [hunk 2]
```diff
@@ -122,21 +124,34 @@ namespace Registry {
...
          : BaseItem{ internalName }, items{ std::move( items_ ) }
       {}
       GroupItem( const GroupItem& ) PROHIBITED;
       ~GroupItem() override = 0;
 
-      // Whether the item is non-significant for path naming
-      // when it also has an empty name
-      virtual bool Transparent() const = 0;
+      //! Choose treatment of the children of the group when merging trees
+      enum Ordering {
```
Comment (saintmatthieu) [inline libraries/lib-registries/Registry.h:142]: Have you considered using `enum class` ? One wouldn't get false positives searching the code for 
e.g. `Ordering::Strong` as opposed to just `Strong`, for example.

File: src/Menus.cpp [hunk 3]
```diff
@@ -103,46 +103,50 @@ void MenuManager::UpdatePrefs()
    mStopIfWasPaused = true;  // not configurable for now, but could be later.
 }
 
-void MenuVisitor::BeginGroup( Registry::GroupItem &item, const Path &path )
+void MenuVisitor::BeginGroup( Registry::GroupItemBase &item, const Path &path )
```
Comment (saintmatthieu) [inline src/Menus.cpp:106]: It was confusing to find the declaration of this guy not in Menus.h but CommandManager.h

**Review by saintmatthieu — COMMENTED** [review]
The code is unfortunately too dense for me to make sense of in a half a day. If more attention is requested from me to this PR please let me know, otherwise I might leave it at that and leave only these superficial comments.

File: src/Menus.cpp (hunk 3 above)
Comment (Paul-Licameli) [reply src/Menus.cpp:106]: Yes:  Menus and CommandManager are tangled up badly, and there is another big pull request in progress to sort that out.

**Pull request merged**